src/lib.rs

```diff
@@ -5337,6 +5361,12 @@ mod test {
     }
 
     #[test]
+    fn fn_with_arguments_with_box_pattern() {
```

Comment:
I think it's safe to remove this test; we already cover that functions use the general set of "pattern" above. The presence of this test makes it look like there's something extra-special about box patterns in function arguments, which there isn't.